src/site/antora/modules/ROOT/pages/manual/configuration.adoc

```diff
@@ -14,1856 +14,1019 @@
-will look for `log4j2.yaml` or `log4j2.yml` on the classpath.
-8.  If a YAML file cannot be located the JSON ConfigurationFactory will
-look for `log4j2.json` or `log4j2.jsn` on the classpath.
-9.  If a JSON file cannot be located the XML ConfigurationFactory will
-try to locate `log4j2.xml` on the classpath.
-10. If no configuration file could be located the `DefaultConfiguration`
-will be used. This will cause logging output to go to the console.
-
-An example application named `MyApp` that uses log4j can be used to
-illustrate how this is done.
-
-[source,java]
-----
-import com.foo.Bar;
-
-// Import log4j classes.
-import org.apache.logging.log4j.Logger;
-import org.apache.logging.log4j.LogManager;
-
-public class MyApp {
-
-    // Define a static logger variable so that it references the
-    // Logger instance named "MyApp".
-    private static final Logger logger = LogManager.getLogger(MyApp.class);
-
-    public static void main(final String... args) {
-
-        // Set up a simple configuration that logs on the console.
-
-        logger.trace("Entering application.");
-        Bar bar = new Bar();
-        if (!bar.doIt()) {
-            logger.error("Didn't do it.");
-        }
-        logger.trace("Exiting application.");
-    }
-}
-----
+Log4j Core can be configured in two ways:
 
-`MyApp` begins by importing log4j related classes. It then defines a
-static logger variable with the name `MyApp` which happens to be the
-fully qualified name of the class.
+* through a <<configuration-file>>.
+Since version 2.0 the configuration file format is considered as part of the public API, and it remains stable even across major version upgrades.
 
-`MyApp` uses the `Bar` class defined in the package`com.foo`.
+* through xref:manual/customconfig.adoc[Programmatic Configuration], which provides a larger spectrum of possible customizations, but might require code changes even for minor version upgrades, according to https://semver.org/[semantic versioning] rules.
 
-[source,java]
-----
-package com.foo;
-import org.apache.logging.log4j.Logger;
-import org.apache.logging.log4j.LogManager;
+[NOTE]
+====
+To prevent a chicken-and-egg problem some configuration options, such as the location of the configuration file can be supplied only through xref:manual/systemproperties.adoc[System properties].
+====
 
-public class Bar {
-  static final Logger logger = LogManager.getLogger(Bar.class.getName());
+[#configuration-file]
+== Configuration file
 
-  public boolean doIt() {
-    logger.entry();
-    logger.error("Did it again!");
-    return logger.exit(false);
-  }
-}
-----
+TIP: For a quick example of configuration file see the xref:manual/installation.adoc#impl-core-config[Configuring Log4j Core section].
 
-Log4j will provide a default configuration if it cannot locate a
-configuration file. The default configuration, provided in the
-DefaultConfiguration class, will set up:
+Log4j Core can be configured using multiple configuration file formats.
```

Comment:
Point taken, I removed the reference in be54ca2aefd708b8159548277d99e12aea948980, since it is too soon to give any configuration file examples.

Those start in the "Syntax" section.